validations/mta_validate.go

```diff
@@ -14,6 +14,67 @@ import (
 	"github.com/SAP/cloud-mta/mta"
 )
 
+const (
+	SeverityError   = "error"
+	SeverityWarning = "warning"
+)
+
+type ValidationResult map[string][]FileValidationIssue
+type FileValidationIssue struct {
+	// Severity - the severity of the message. Possible values: "error", "warning".
+	Severity string `json:"severity"`
+	// Message - the validation message
+	Message string `json:"message"`
+	// Line - the line number of the issue
+	Line int `json:"line"`
+	// Column - the column number of the issue
+	Column int `json:"column"`
+}
+
+// Validate validates an mta.yaml file and a list of mta extension files, and returns the issues for each file.
+func Validate(mtaPath string, extensions []string) (ValidationResult, []string, error) {
+	allIssues := make(ValidationResult)
+	// Assuming the project path is the folder of the mta.yaml
+	projectPath := filepath.Dir(mtaPath)
+	mtaYamlFileName := filepath.Base(mtaPath)
+	warningIssues, errorIssues, e := validateMtaYaml(projectPath, mtaYamlFileName, true, true, true, "")
+	if e != nil {
+		errorIssues = appendIssue(errorIssues, e.Error(), 0, 0)
+	}
+	allIssues[mtaPath] = createFileIssues(warningIssues, errorIssues)
+
+	for _, extPath := range extensions {
+		warningIssues, errorIssues, e = validateMtaext(projectPath, extPath, true, true, true, "")
+		if e != nil {
+			errorIssues = appendIssue(errorIssues, e.Error(), 0, 0)
+		}
+		allIssues[extPath] = createFileIssues(warningIssues, errorIssues)
+	}
+
+	_, _, e = mta.GetMtaFromFile(mtaPath, extensions, true)
+	if e != nil {
+		// Ignore errors which are not on a specific extension (if they are on the mta.yaml we already got them earlier)
+		// and parse errors from extensions (we already got them earlier too)
+		if extErr, ok := e.(*mta.ExtensionError); ok && !extErr.IsParseError {
+			allIssues[extErr.FileName] = append(allIssues[extErr.FileName], FileValidationIssue{SeverityError, e.Error(), 0, 0})
+		}
+	}
+
+	return allIssues, nil, nil
```

Comment:
I will remove the other 2 parameters (it was because this function's return values are passed to another function, but I will use `nil` there instead)

Reply:
Done